cpp/tests/unary/unary_ops_test.cu

```diff
@@ -479,6 +529,61 @@ TEST_F(CastTimestamps, IsIdempotent) {
       timestamps_ns, *timestamps_ns_got);
 }
 
+template <typename T>
+struct CastTimestampsTyped : public cudf::test::BaseFixture {};
+
+TYPED_TEST_CASE(CastTimestampsTyped, cudf::test::TimestampTypes);
+
+// Return a list of timestamp type ids whose precision is greater than or equal
+// to the input type id
+std::vector<cudf::type_id> get_higher_precision_timestamp_type_ids(
+    cudf::type_id search) {
+  bool found = false;
+  std::vector<cudf::type_id> gte_ids{};
+  std::vector<cudf::type_id> timestamp_ids{
+      cudf::TIMESTAMP_DAYS, cudf::TIMESTAMP_SECONDS,
+      cudf::TIMESTAMP_MILLISECONDS, cudf::TIMESTAMP_MICROSECONDS,
+      cudf::TIMESTAMP_NANOSECONDS};
+  for (cudf::type_id type_id : timestamp_ids) {
+    if (found || (found = (type_id == search))) {
+      gte_ids.emplace_back(type_id);
+    }
+  }
+  return gte_ids;
+}
+
+// Test that all timestamps whose precision is >= to the TypeParam
+// down-casts appropriately to the lower-precision TypeParam
+TYPED_TEST(CastTimestampsTyped, DownCastingTruncatesValues) {
```

Comment:
Most of these changes are from clang-format. This line is where the new tests begin.